docs/2.7.0/docs/daml-finance/tutorials/instrument-extension-packages/fixed-rate-bond.rst

```diff
@@ -38,29 +41,53 @@ The code for this tutorial can be executed via the ``runFixedRateBond`` function
+pay a constant coupon each coupon period. The coupon is quoted on a yearly basis
+(per annum, p.a.), but it could be paid more frequently.
+As an example we will create a bond instrument paying a 1.1% p.a. coupon with a 12M coupon period.
+We start by creating our instrument. Here are the key terms:
 
 .. literalinclude:: ../../finance-instruments/daml/Scripts/FixedRateBond.daml
   :language: daml
   :start-after: -- CREATE_FIXED_RATE_BOND_VARIABLES_BEGIN
   :end-before: -- CREATE_FIXED_RATE_BOND_VARIABLES_END
 
-We also need holiday calendars, which are used to adjust non-business days:
+The :ref:`day count convention <type-daml-finance-interface-types-date-daycount-daycountconventionenum-67281>`
+is used to determine how many days, i.e., what fraction of a full year, each coupon period has. This
+will determine the exact coupon amount that will be paid each period.
+
+The :ref:`business day convention <type-daml-finance-interface-types-date-calendar-businessdayconventionenum-88986>`
+determines *how* a coupon date is adjusted if it falls on a non-business day.
+
+We also need holiday calendars, which determine *when* to adjust dates:
 
 .. literalinclude:: ../../finance-instruments/daml/Scripts/FixedRateBond.daml
   :language: daml
   :start-after: -- CREATE_FIXED_RATE_BOND_CALENDARS_BEGIN
   :end-before: -- CREATE_FIXED_RATE_BOND_CALENDARS_END
 
-The coupon dates and the redemption date are described using a periodic schedule. This also allows
-us to define a stub period (if required):
+The coupon dates and the redemption date are described using a
+:ref:`PeriodicSchedule <constr-daml-finance-interface-types-date-schedule-periodicschedule-99705>`:
 
 .. literalinclude:: ../../finance-instruments/daml/Scripts/FixedRateBond.daml
   :language: daml
   :start-after: -- CREATE_PERIODIC_SCHEDULE_BEGIN
   :end-before: -- CREATE_PERIODIC_SCHEDULE_END
 
-We can now create the bond instrument using a factory:
+This is used to determine the periods that are used to calculate the coupon. There are a few things
+to note here:
+
+- The :ref:`RollConventionEnum <type-daml-finance-interface-types-date-rollconvention-rollconventionenum-73360>`
+  defines whether dates are rolled at the end of the month or on a given date of the month. In our
+  example above, we went for the latter option.
+- The :ref:`StubPeriodTypeEnum <type-daml-finance-interface-types-date-schedule-stubperiodtypeenum-69372>`
+  allows you to explicitly specify what kind of stub period the bond should have. This is optional
+  and not used in the example above. Instead, we defined the stub implicitly by specifying a
+  ``firstRegularPeriodStartDate``: since the time between the issue date and the first regular
+  period start date is less than 12M (our regular coupon period), this implies a short initial
+  stub period.
+
+Now that we have defined the terms we can create the bond instrument using a factory:
```

Comment:
Just to confirm, the below snippet is called CREATE_FIXED_RATE_BOND_HOLDING_BEGIN. Does it also create the instrument?

Reply:
No, the instrument is created earlier, in a separate snippet. Here is a preview on how it looks: https://docs.daml.com/2.7.0/daml-finance/tutorials/instrument-extension-packages/fixed-rate-bond.html

Reply:
Sorry, I got confused by Github and didn't realize there was some code in between ... 😓 haven't fully woken up yet

Reply:
No problem :)
All good to approve?